### PR TITLE
Fix creation of tasks with Git repositories via the SDK

### LINKED ISSUE #4365: Create task with git repository
### My actions before raising this issue
- [x] Read/searched [the docs](https://github.com/opencv/cvat/tree/master#documentation)
- [x] Searched [past issues](/issues)

### Expected Behaviour 
Create a task with github repository for synchronizing task https://openvinotoolkit.github.io/cvat/docs/manual/advanced/task-synchronization/

### Context

I'm trying to create new task with a git repository using cli https://openvinotoolkit.github.io/cvat/docs/manual/advanced/cli// but it always shows error that "format" key is missing. When i dont create task with github repository, everything works fine. 

### Your Environment

- Code example or link to GitHub repo or gist to reproduce problem:
```
subprocess.call([
            'python', 'cvat/utils/cli/cli.py', \
            "--auth", auth_account, \
            "create", task_name, \
            "--labels", '[{"name": "person"}]', \
            "--project_id", str(args.project_id), \
            "--dataset_repository_url", "https://github.com/LeDuySon/BackupDataset", \
            "--annotation_format", "CVAT 1.1", \
            "local", video_path
            ])
```
- Other diagnostic information / logs:
image
image

### PR DESCRIPTION
**Pull request by SpecLad**
### Motivation and context

Fixes #4365.

### How has this been tested?

SDK unit tests + I did some Git-related operations via the UI to make sure they weren't broken.

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added a description of my changes into [CHANGELOG](https://github.com/cvat-ai/cvat/blob/develop/CHANGELOG.md) file
- ~~[ ] I have updated the [documentation](
  https://github.com/cvat-ai/cvat/blob/develop/README.md#documentation) accordingly~~
- [x] I have added tests to cover my changes
- [x] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- ~~[ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))~~

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

**Comment by bsekachev** [comment]
Fix for failed 3D test: https://github.com/opencv/cvat/pull/5407/commits/3f097f5238dba2962d2fcd17ccc749323feeabc2
Explanation: https://github.com/opencv/cvat/pull/5407#issuecomment-1334326304

**Review by zhiltsov-max — COMMENTED** [review]
The PR looks good to me. Maybe, we could add some info about the expected format in the CLI output and in the task creating function docs. A link to the [description page](https://opencv.github.io/cvat/docs/manual/basics/creating_an_annotation_task/#dataset-repository) may be enough for this .

**Comment by SpecLad** [comment]
> Maybe, we could add some info about the expected format in the CLI output and in the task creating function docs.

Maybe, but that's not related to the bug. 🙂 I didn't change the "face" of the API, so I don't think documenting it should be part of this PR.

File: tests/git_server/entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
```
Comment (zhiltsov-max) [inline tests/git_server/entrypoint.sh:1]: Is there any reason to make this file owned by root with the 755 rights? It's uncomfortable to work with.

Comment (SpecLad) [reply tests/git_server/entrypoint.sh:1]: I don't understand what you mean. Git doesn't record ownership or file permissions (beyond executable/non-executable); this is just a normal executable file.

Comment (zhiltsov-max) [reply tests/git_server/entrypoint.sh:1]: I get the following problem: when I change the branch, the file is missing and shown as removed in git, and I get the following error:

```
$ git checkout tests/git_server/entrypoint.sh
error: unable to create file tests/git_server/entrypoint.sh: Permission denied
Updated 1 path from the index
```

**Review by zhiltsov-max — APPROVED** [review]

**Pull request merged**